From 08ed594ed0b18afbadad4cf2bede9b0420278f3e Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Thu, 10 Jul 2014 01:01:53 -0400
Subject: [acpi] hotplug/pci: Drop flags field from struct acpiphp_bridge

Message-id: <20140710010153.22146.90122.stgit@gir.stowe>
Patchwork-id: 85574
O-Subject: [RHEL7.1 PATCH 035/105] ACPI / hotplug / PCI: Drop flags field from struct acpiphp_bridge
Bugzilla: 1114228
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1114228
Upstream Status: 75a33ed1b58005e455cb6533a7689ac0eb6bedd6

commit 75a33ed1b58005e455cb6533a7689ac0eb6bedd6
Author: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Date:   Sat Jul 13 23:27:25 2013 +0200

    ACPI / hotplug / PCI: Drop flags field from struct acpiphp_bridge

    The only bridge flag used by the ACPI-based PCI hotplug (ACPIPHP)
    code is BRIDGE_HAS_EJ0, but it is only used by the event handling
    function hotplug_event() and if that flag is set, the corresponding
    function flag FUNC_HAS_EJ0 is set as well, so that bridge flag is
    redundant.

    For this reason, drop BRIDGE_HAS_EJ0 and all code referring to it
    and since it is the only bridge flag defined, drop the flags field
    from struct acpiphp_bridge entirely.

    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
    Tested-by: Mika Westerberg <mika.westerberg@linux.intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/pci/hotplug/acpiphp.h b/drivers/pci/hotplug/acpiphp.h
index 23c308f..0663493 100644
--- a/drivers/pci/hotplug/acpiphp.h
+++ b/drivers/pci/hotplug/acpiphp.h
@@ -73,8 +73,6 @@ struct acpiphp_bridge {
 
  int nr_slots;
 
- u32 flags;
-
  /* This bus (host bridge) or Secondary bus (PCI-to-PCI bridge) */
  struct pci_bus *pci_bus;
 
@@ -144,9 +142,6 @@ struct acpiphp_attention_info
 /* ACPI _STA method value (ignore bit 4; battery present) */
 #define ACPI_STA_ALL   (0x0000000f)
 
-/* bridge flags */
-#define BRIDGE_HAS_EJ0  (0x00000001)
-
 /* slot flags */
 
 #define SLOT_POWEREDON  (0x00000001)
diff --git a/drivers/pci/hotplug/acpiphp_glue.c b/drivers/pci/hotplug/acpiphp_glue.c
index 744c72b..fe09703 100644
--- a/drivers/pci/hotplug/acpiphp_glue.c
+++ b/drivers/pci/hotplug/acpiphp_glue.c
@@ -986,9 +986,6 @@ static void hotplug_event(acpi_handle handle, u32 type, void *data)
  case ACPI_NOTIFY_EJECT_REQUEST:
   /* request device eject */
   pr_debug("%s: Device eject notify on %s\n", __func__, objname);
-  if (bridge && !(bridge->flags & BRIDGE_HAS_EJ0))
-   break;
-
   if (!(acpiphp_disable_slot(func->slot)))
    acpiphp_eject_slot(func->slot);
 
@@ -1125,12 +1122,6 @@ void acpiphp_enumerate_slots(struct pci_bus *bus)
   mutex_unlock(&acpiphp_context_lock);
  }
 
- status = acpi_get_handle(bridge->handle, "_EJ0", &handle);
- if (ACPI_SUCCESS(status)) {
-  pr_debug("found ejectable p2p bridge\n");
-  bridge->flags |= BRIDGE_HAS_EJ0;
- }
-
  /* must be added to the list prior to calling register_slot */
  mutex_lock(&bridge_mutex);
  list_add(&bridge->list, &bridge_list);
-- 
1.7.1